src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,375 @@
+                            final CheckFlag flag = gson.fromJson(line, CheckFlag.class);
+
+                            this.batchFlagStatement(flagSqlStatement, flag);
+
+                            StreamSupport.stream(features.spliterator(), false).forEach(
+                                    feature -> this.batchFlagFeatureStatement(featureSqlStatement,
+                                            flag, feature.getAsJsonObject()));
+                        });
+
+                        flagSqlStatement.executeBatch();
+                        featureSqlStatement.executeBatch();
+
+                        startIndex = startIndex + BATCH_SIZE;
+                    }
+                    while (endIndex != lines.size() - 1 && startIndex < lines.size());
+
+                    featureSqlStatement.close();
+                    flagSqlStatement.close();
+
+                }
+                catch (final IOException error)
+                {
+                    logger.error("Exception while reading {}:", file, error);
+                }
+                catch (final SQLException error)
+                {
+                    logger.error("Exception batch executing flag statements", error);
+                }
+            });
+        });
+
+        logger.info("Atlas Checks database upload command finished in {} seconds.",
+                timer.elapsedSince().asSeconds());
+
+        return 0;
+    }
+
+    @Override
+    public String getCommandName()
+    {
+        return "flag-database";
+    }
+
+    @Override
+    public String getSimpleDescription()
+    {
+        return "Upload Atlas Checks flags into a Postgres database";
+    }
+
+    @Override
+    public void registerManualPageSections()
+    {
+        this.addManualPageSection("DESCRIPTION", FlagDatabaseSubCommand.class
+                .getResourceAsStream("FlagDatabaseSubCommandDescriptionSection.txt"));
+        this.addManualPageSection("EXAMPLES", FlagDatabaseSubCommand.class
+                .getResourceAsStream("FlagDatabaseSubCommandExamplesSection.txt"));
+    }
+
+    @Override
+    public void registerOptionsAndArguments()
+    {
+        this.registerOptionWithRequiredArgument(FLAG_PATH_INPUT, 'f',
+                "A directory of folders containing atlas-checks log files.",
+                OptionOptionality.REQUIRED, FLAG_PATH_INPUT);
+        this.registerOptionWithRequiredArgument(DATABASE_URL_INPUT, 't',
+                "Database connection string", OptionOptionality.REQUIRED, DATABASE_URL_INPUT);
+        this.registerOptionWithRequiredArgument(COUNTRIES_INPUT, 'i',
+                "List of iso3 country codes to filter by", OptionOptionality.OPTIONAL,
+                COUNTRIES_INPUT);
+        this.registerOptionWithRequiredArgument(CHECKS_INPUT, 'c', "A folder to output results to.",
+                OptionOptionality.OPTIONAL, CHECKS_INPUT);
+        super.registerOptionsAndArguments();
+    }
+
+    /**
+     * @param sql
```

Comment:
Could use a description.

Reply:
comment is below - "PreparedStatement to add parameterized values to"

Reply:
Sorry, I meant the function as a whole, not the parameter.

Reply:
Ohhh yeah ok